### ISSUE TITLE
No implicit conversion of nil into String

### ISSUE THREAD

**Issue by leighhalliday** [yob/pdf-reader#169]
I am trying to read the text from a PDF... unfortunately I can't share the PDF because it contains banking information but I'll do my best to set up the context.

``` ruby
reader = PDF::Reader.new("~/file.pdf")
page = reader.pages.first
page.text
```

Produces:

```
TypeError: no implicit conversion of nil into String
from /Users/patborders/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/bundler/gems/pdf-reader-df10bcedffbb/lib/pdf/reader/buffer.rb:235:in `prepare_inline_token'
```

**Comment by yob** [comment]
Thanks for the detailed write up and new spec.

I agree that the conditional isn't ideal, but I lean towards improving support for more files, adding a spec, and refactoring later if we feel so inclined.

**Comment by pwood** [comment]
Just come across this ticket, I too have PDFs from a large bank. I believe the fix provided isn't 100% correct. It assumes that the byte before the EI will be either \s or 0x00, however I have a PDF which ends 0x1F on an embedded image.

I'd posit that the when there is no space the last byte is part of the image rather than a delimiter and the provided fix works under a very specific use case.

While attempting to fix it I was using this spec.

```
  it "should correctly tokenise an inline image with no whitespace before the letters 'EI' but instead a none 0x00 value" do
    io = StringIO.new(binary_string("BI ID aaa bbb ccc \xF0\xF0\xF0\x01EI"))
    buf = PDF::Reader::Buffer.new(io, :content_stream => true)

    expect(buf.pos).to eql(0)
    expect(buf.token).to eql("BI")
    expect(buf.token).to eql("ID")
    expect(buf.token).to eql(binary_string("aaa bbb ccc \xF0\xF0\xF0\x01"))
    expect(buf.token).to eql("EI")
  end
```

I haven't managed to create a suitable fix yet (still learning about PDF internals at the same time) but I wanted to at least bring it up in case someone else encounters it.

**Comment by yob** [comment]
Thanks for the great research.

I'm relatively unfamiliar with inline images, but what you're proposing sounds plausible. We've merged the initial fix in #172, and I'll happily accept a refinement PR if you can come up with something.
